### PR TITLE
Feat tensorboard

### LINKED ISSUE #1811: Improvement proposal for TensorBoard integration
## Motivation

- Improve the display of TensorBoard

### 1. Trial ID

I guess using directory name as trial ID is helpful to find a good trial.
Current integration uses default hash.

**Current display**
スクリーンショット 2020-09-12 10 17 18

**Proposal**
スクリーンショット 2020-09-12 10 08 48

### 2. Use trial number as horizontal axis

**Current display**
スクリーンショット 2020-09-12 10 17 32

**Proposal**
スクリーンショット 2020-09-12 10 05 14

### PR DESCRIPTION
**Pull request by sfujiwara**
# Motivation

This PR close #1811.

### Major changes

- Use `trial-{trial number}` as IDs displayed on TensorBoard
- Use trial number as horizontal axis on scalar tab

### Minor changes

Add description of metric:
image

Add hparams config as below.
We can control metrics displayed on HPARAMS tab and it might be useful in future.

```python
hp.hparams_config(
    hparams=list(hparams.keys()),
    # Control metric values displayed on HPARAMS tab of TensorBoard.
    metrics=[
        hp.Metric(
            tag=self._metric_name,
            display_name=self._metric_name,
            description="Trial value optimized with Optuna."
        )
    ]
)
```

### Examples

You can see the result of sample code here:

**The result of [Current example (Optuna v2.1.0)](https://github.com/optuna/optuna/blob/release-v2.1.0/examples/tensorboard_simple.py):**
https://tensorboard.dev/experiment/fN5hPkJlSKiXcxePCDUosw/

**The proposed change:**
https://tensorboard.dev/experiment/GaKU5NcjT1CweWO6BzXDqg/

### CODE REVIEW

File: optuna/integration/tensorboard.py [hunk 1]
```diff
@@ -49,8 +49,19 @@ def __call__(self, study: optuna.study.Study, trial: optuna.trial.FrozenTrial) -
...
-            hp.hparams(hparams)  # record the values used in this trial
-            tf.summary.scalar(self._metric_name, trial_value, step=1)
+            hp.hparams_config(
+                hparams=list(hparams.keys()),
+                # Control metric values displayed on HPARAMS tab of TensorBoard.
+                metrics=[
+                    hp.Metric(
+                        tag=self._metric_name,
+                        display_name=self._metric_name,
+                        description="Trial value optimized with Optuna.",
```
Comment (g-votte) [inline optuna/integration/tensorboard.py:59]: Though `description` is an interesting feature, it looks redundant just to show the same message. How about removing `description` so far?

**Review by g-votte — COMMENTED** [review]
Thanks a lot for your contribution! I put a minor comment.

File: optuna/integration/tensorboard.py (hunk 1 above)
Comment (sfujiwara) [reply optuna/integration/tensorboard.py:59]: OK. Which do you think better?

- Remove `hp.hparams_config`
- Remove only `description` option

Comment (toshihikoyanase) [reply optuna/integration/tensorboard.py:59]: In my understanding, users will get the same output with or without `hp.params_config` if we remove `description` option. If so, I prefer shorter code.

Comment (g-votte) [reply optuna/integration/tensorboard.py:59]: I also prefer shorter code.

@sfujiwara 
Would you mind removing `hp.hparams_config`?

Comment (sfujiwara) [reply optuna/integration/tensorboard.py]: No problem. I removed `hp.hparams_config`!
Removing `hp.hparams_config` does not change the result.
You can see the result is here: https://tensorboard.dev/experiment/MlImIJjdQaOX0gEM774BEA/

**Review by toshihikoyanase — APPROVED** [review]
Thank you for your update. LGTM!

**Review by g-votte — APPROVED** [review]
LGTM!

**Pull request merged**